// @generated by protoc-gen-es v2.0.0 with parameter "target=ts,import_extension=js,json_types=true"
// @generated from file exception/v1/exception.proto (package sourcetool.exception.v1, syntax proto3)
/* eslint-disable */

import type { GenFile, GenMessage } from "@bufbuild/protobuf/codegenv1";
import { fileDesc, messageDesc } from "@bufbuild/protobuf/codegenv1";
import type { Message } from "@bufbuild/protobuf";

/**
 * Describes the file exception/v1/exception.proto.
 */
export const file_exception_v1_exception: GenFile = /*@__PURE__*/
  fileDesc("ChxleGNlcHRpb24vdjEvZXhjZXB0aW9uLnByb3RvEhdzb3VyY2V0b29sLmV4Y2VwdGlvbi52MSJACglFeGNlcHRpb24SDQoFdGl0bGUYASABKAkSDwoHbWVzc2FnZRgCIAEoCRITCgtzdGFja190cmFjZRgDIAMoCUL0AQobY29tLnNvdXJjZXRvb2wuZXhjZXB0aW9uLnYxQg5FeGNlcHRpb25Qcm90b1ABWkVnaXRodWIuY29tL3RyeXNvdXJjZXRvb2wvc291cmNldG9vbC9wcm90by9nby9leGNlcHRpb24vdjE7ZXhjZXB0aW9udjGiAgNTRViqAhdTb3VyY2V0b29sLkV4Y2VwdGlvbi5WMcoCGFNvdXJjZXRvb2xcRXhjZXB0aW9uX1xWMeICJFNvdXJjZXRvb2xcRXhjZXB0aW9uX1xWMVxHUEJNZXRhZGF0YeoCGVNvdXJjZXRvb2w6OkV4Y2VwdGlvbjo6VjFiBnByb3RvMw");

/**
 * @generated from message sourcetool.exception.v1.Exception
 */
export type Exception = Message<"sourcetool.exception.v1.Exception"> & {
  /**
   * @generated from field: string title = 1;
   */
  title: string;

  /**
   * @generated from field: string message = 2;
   */
  message: string;

  /**
   * @generated from field: repeated string stack_trace = 3;
   */
  stackTrace: string[];
};

/**
 * JSON type for the message sourcetool.exception.v1.Exception.
 */
export type ExceptionJson = {
  /**
   * @generated from field: string title = 1;
   */
  title?: string;

  /**
   * @generated from field: string message = 2;
   */
  message?: string;

  /**
   * @generated from field: repeated string stack_trace = 3;
   */
  stackTrace?: string[];
};

/**
 * Describes the message sourcetool.exception.v1.Exception.
 * Use `create(ExceptionSchema)` to create a new message.
 */
export const ExceptionSchema: GenMessage<Exception, ExceptionJson> = /*@__PURE__*/
  messageDesc(file_exception_v1_exception, 0);

